diff -urN MPlayer-0.50/etc/codecs.conf MPlayer-0.50.new/etc/codecs.conf
--- MPlayer-0.50/etc/codecs.conf	Sat Oct  6 13:30:07 2001
+++ MPlayer-0.50.new/etc/codecs.conf	Tue Oct  9 23:41:48 2001
@@ -4,6 +4,77 @@
 ;                   VIDEO CODECS
 ;=============================================================================
 
+; =============== FFMPEG MPEG CODECS =========================
+
+videocodec ffmjpeg
+  info "FFmpeg MJPEG decoder"
+  status working
+  fourcc MJPG,mjpg
+  driver ffmpeg
+  dll mjpeg
+;  out YUY2
+  out YV12
+
+videocodec ffi263
+  info "FFmpeg I263 decoder"
+  status working
+  fourcc I263,i263
+  driver ffmpeg
+  dll h263i
+  out YV12
+
+videocodec ffh263
+  info "FFmpeg H263+ decoder"
+  status working
+  fourcc H263,U263
+  driver ffmpeg
+  dll h263
+  out YV12
+
+videocodec ffdivx
+  info "FFmpeg DivX ;-) (MS MPEG-4 v3)"
+  status working
+  fourcc MPG3,mpg3 div3
+  fourcc MP43,mp43 div3   ; M$ MPEG4 v3   (fourcc mapping to div3)
+  fourcc DIV5,div5 div3   ; DivX 3.20 
+  fourcc DIV6,div6 div4   ;   -||-
+  fourcc DIV3,div3,DIV4,div4
+  fourcc AP41 div3	  ; AngelPotion stuff
+  driver ffmpeg
+  dll msmpeg4
+  out YV12
+
+videocodec ffodivx
+  info "FFmpeg OpenDivX"
+  status working
+;  comment "doesn't support >= 4.0alpha48 movies yet"
+  fourcc DIVX,divx
+  fourcc DIV1,div1 divx
+  fourcc MP4S,mp4s	; ISO MPEG-4 Video V1
+  format 0x4
+  driver ffmpeg
+  dll mpeg4 ;opendivx
+  out YV12
+
+videocodec ffmpeg12
+  info "FFmpeg-1"
+  status working
+  format 0x10000001  ; mpeg 1
+  format 0x10000002  ; mpeg 2
+  driver ffmpeg
+  dll "mpegvideo"
+  out YV12
+
+videocodec mpegpes
+  info "Mpeg PES output (.mpg or DVB card)"
+  comment "for hardware decoding"
+  status working
+  format 0x10000001  ; mpeg 1
+  format 0x10000002  ; mpeg 2
+  driver libmpeg2
+  dll "libmpeg2"
+  out MPES
+
 ; =============== Win32/libmpeg2/ProjectMayo MPEG CODECS =====================
 
 videocodec u263
@@ -21,6 +92,16 @@
 ;  out YUY2 flip
   out BGR24 flip
 
+videocodec mpeg12
+  info "MPEG 1 or 2"
+  comment "with postprocessing"
+  status working
+  format 0x10000001  ; mpeg 1
+  format 0x10000002  ; mpeg 2
+  driver libmpeg2
+  dll "libmpeg2"
+  out YV12
+
 videocodec divxds
   info "DivX ;-) (MS MPEG-4 v3)"
   comment "with postprocessing"
@@ -37,16 +118,6 @@
   out YUY2
   out BGR32,BGR24,BGR16,BGR15
 
-videocodec mpeg12
-  info "MPEG 1 or 2"
-  comment "with postprocessing"
-  status working
-  format 0x10000001  ; mpeg 1
-  format 0x10000002  ; mpeg 2
-  driver libmpeg2
-  dll "libmpeg2"
-  out YV12
-
 videocodec mpeg4
   info "Microsoft MPEG-4 v1/v2"
   status working
@@ -84,7 +155,6 @@
   dll "divxc32.dll"
   out YUY2 yuvhack
   out BGR32,BGR24,BGR15
-
   
 videocodec odivx
   info "OpenDivX (MPEG-4 v2)"
@@ -155,81 +225,8 @@
   out UYVY
   out BGR32,BGR24,BGR15
 
-; =============== FFMPEG MPEG CODECS =========================
-
-videocodec ffmjpeg
-  info "FFmpeg MJPEG decoder"
-  status working
-  fourcc MJPG,mjpg
-  driver ffmpeg
-  dll mjpeg
-;  out YUY2
-  out YV12
-
-videocodec ffi263
-  info "FFmpeg I263 decoder"
-  status working
-  fourcc I263,i263
-  driver ffmpeg
-  dll h263i
-  out YV12
-
-videocodec ffh263
-  info "FFmpeg H263+ decoder"
-  status working
-  fourcc H263,U263
-  driver ffmpeg
-  dll h263
-  out YV12
-
-videocodec ffdivx
-  info "FFmpeg DivX ;-) (MS MPEG-4 v3)"
-  status working
-  fourcc MPG3,mpg3 div3
-  fourcc MP43,mp43 div3   ; M$ MPEG4 v3   (fourcc mapping to div3)
-  fourcc DIV5,div5 div3   ; DivX 3.20 
-  fourcc DIV6,div6 div4   ;   -||-
-  fourcc DIV3,div3,DIV4,div4
-  fourcc AP41 div3	  ; AngelPotion stuff
-  driver ffmpeg
-  dll msmpeg4
-  out YV12
-
-videocodec ffodivx
-  info "FFmpeg OpenDivX"
-  status working
-;  comment "doesn't support >= 4.0alpha48 movies yet"
-  fourcc DIVX,divx
-  fourcc DIV1,div1 divx
-  fourcc MP4S,mp4s	; ISO MPEG-4 Video V1
-  format 0x4
-  driver ffmpeg
-  dll mpeg4 ;opendivx
-  out YV12
-
-videocodec ffmpeg12
-  info "FFmpeg-1"
-  status working
-  format 0x10000001  ; mpeg 1
-  format 0x10000002  ; mpeg 2
-  driver ffmpeg
-  dll "mpegvideo"
-  out YV12
-
-
-videocodec mpegpes
-  info "Mpeg PES output (.mpg or DVB card)"
-  comment "for hardware decoding"
-  status working
-  format 0x10000001  ; mpeg 1
-  format 0x10000002  ; mpeg 2
-  driver libmpeg2
-  dll "libmpeg2"
-  out MPES
-
 ; ============== OTHER STUFF =======================
 
-
 videocodec vp3
   info "On2 OpenSource VP3-Codec"
   status working
@@ -404,7 +401,6 @@
   out YUY2
   out BGR32,BGR24,BGR15
 
-
 videocodec m261
   info "M261"
   status untested
@@ -473,21 +469,6 @@
   driver pcm
   dll "uncompressed"
 
-audiocodec divx
-  info "DivX audio (WMA)"
-  status working
-  format 0x160
-  format 0x161
-  driver acm
-  dll "divxa32.acm"
-
-audiocodec msadpcm
-  info "MS ADPCM"
-  status working
-  format 0x2
-  driver acm
-  dll "msadp32.acm"
-
 ; now mp3lib is stable enough!?
 audiocodec mp3
   info "MPEG layer-2, layer-3"
@@ -508,6 +489,61 @@
   dll "mpegaudio"
   flags seekable
 
+audiocodec alaw
+  info "aLaw"
+  status working
+  format 0x6
+  driver alaw
+  dll "alaw"
+
+audiocodec ulaw
+  info "uLaw"
+  status working
+  format 0x7
+  driver alaw
+  dll "ulaw"
+
+audiocodec dvdpcm
+  info "Uncompressed DVD PCM"
+  status working
+  format 0x10001
+  driver dvdpcm
+  dll "uncompressed"
+
+audiocodec ac3
+  info "AC3"
+  status working
+  format 0x2000
+  driver libac3
+  dll "libac3"
+
+audiocodec vorbis
+  info "OggVorbis Audio Decoder"
+  status working
+  comment "OggVorbis driver using libvorbis"
+  format 0xFFFE
+  driver libvorbis
+  dll "libvorbis"
+  flags seekable
+; acm codec doesn't work, haven't tried zorannt dshow codec
+;  driver acm
+;  dll "vorbis.acm"
+
+audiocodec divx
+  info "DivX audio (WMA)"
+  status working
+  format 0x160
+  format 0x161
+  driver acm
+  dll "divxa32.acm"
+
+audiocodec msadpcm
+  info "MS ADPCM"
+  status working
+  format 0x2
+  driver acm
+  dll "msadp32.acm"
+
 audiocodec mp3acm
   info "MPEG layer-3"
   status working
@@ -540,34 +576,6 @@
   driver msgsm
   dll "xa_gsm.c"
 
-audiocodec alaw
-  info "aLaw"
-  status working
-  format 0x6
-  driver alaw
-  dll "alaw"
-
-audiocodec ulaw
-  info "uLaw"
-  status working
-  format 0x7
-  driver alaw
-  dll "ulaw"
-
-audiocodec dvdpcm
-  info "Uncompressed DVD PCM"
-  status working
-  format 0x10001
-  driver dvdpcm
-  dll "uncompressed"
-
-audiocodec ac3
-  info "AC3"
-  status working
-  format 0x2000
-  driver libac3
-  dll "libac3"
-
 audiocodec voxware
   info "VoxWare"
   status working	; audio is bubbling
@@ -599,16 +607,4 @@
   format 0x2000
   driver hwac3
   dll "ac3-iec958.c"
-
-audiocodec vorbis
-  info "OggVorbis Audio Decoder"
-  status working
-  comment "OggVorbis driver using libvorbis"
-  format 0xFFFE
-  driver libvorbis
-  dll "libvorbis"
-  flags seekable
-; acm codec doesn't work, haven't tried zorannt dshow codec
-;  driver acm
-;  dll "vorbis.acm"
-
+  
\ No newline at end of file
