--- MPlayer-1.1/configure~	2012-06-17 23:45:37.000000000 +0300
+++ MPlayer-1.1/configure	2012-06-17 23:47:45.579510002 +0300
@@ -6701,10 +6701,7 @@
              -I$_livelibdir/UsageEnvironment/include \
              -I$_livelibdir/BasicUsageEnvironment/include \
              -I$_livelibdir/groupsock/include"
-    ld_tmp="$_livelibdir/liveMedia/libliveMedia.a \
-            $_livelibdir/UsageEnvironment/libUsageEnvironment.a \
-            $_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a \
-            $_livelibdir/groupsock/libgroupsock.a \
+    ld_tmp="-lliveMedia -lgroupsock -lUsageEnvironment -lBasicUsageEnvironment \
             -lstdc++ $ld_sock"
     test -e "$_livelibdir/BasicUsageEnvironment/libBasicUsageEnvironment.a" &&
         cxx_check $inc_tmp $ld_tmp &&
