--- MPlayer-1.1/configure.orig	2013-02-03 09:17:44.221291917 +0100
+++ MPlayer-1.1/configure	2013-02-03 16:59:43.280714166 +0100
@@ -4914,12 +4914,13 @@
 echocheck "OpenJPEG (JPEG 2000) support"
 if test "$libopenjpeg" = auto ; then
   libopenjpeg=no
-  define_statement_check OPJ_STATIC openjpeg.h 'opj_dparameters_t dec_params; opj_set_default_decoder_parameters(&dec_params);opj_decode_with_info(0,0,0)' -lopenjpeg && libopenjpeg=yes
+  define_statement_check OPJ_STATIC openjpeg.h 'opj_dparameters_t dec_params; opj_set_default_decoder_parameters(&dec_params);opj_decode_with_info(0,0,0)' $(pkg-config --cflags --libs libopenjpeg) && libopenjpeg=yes
 fi
 echores "$libopenjpeg"
 if test "$libopenjpeg" = yes ; then
   def_libopenjpeg='#define CONFIG_LIBOPENJPEG 1'
-  extra_ldflags="$extra_ldflags -lopenjpeg"
+  extra_cflags="$extra_cflags $(pkg-config --cflags libopenjpeg)"
+  extra_ldflags="$extra_ldflags $(pkg-config --libs libopenjpeg)"
   libavdecoders="$libavdecoders LIBOPENJPEG_DECODER"
   libavencoders="$libavencoders LIBOPENJPEG_ENCODER"
   codecmodules="OpenJPEG $codecmodules"
