+++ MPlayer-1.1/configure.orig	2013-01-28 21:25:18.354520436 +0100
--- MPlayer-1.1/configure	2013-01-30 20:06:23.334350773 +0100
@@ -6686,6 +6686,7 @@
 echocheck "LIVE555 Streaming Media libraries"
 if test "$_live" = auto  && test "$networking" = yes ; then
   cat > $TMPCPP << EOF
+#define RTSPCLIENT_SYNCHRONOUS_INTERFACE 1
 #include <liveMedia.hh>
 #if (LIVEMEDIA_LIBRARY_VERSION_INT < 1141257600)
 #error Please upgrade to version 2006.03.03 or later of the "LIVE555 Streaming Media" libraries - available from <www.live555.com/liveMedia/>
