diff -urN MPlayer-0.50/mplayer.c MPlayer-0.50.new/mplayer.c
--- MPlayer-0.50/mplayer.c	Sun Oct  7 02:57:28 2001
+++ MPlayer-0.50.new/mplayer.c	Tue Oct  9 19:09:39 2001
@@ -413,7 +413,7 @@
 {
 char *conffile;
 int conffile_fd;
-if (parse_config_file(conf, "/etc/mplayer.conf") < 0)
+if (parse_config_file(conf, "/etc/mplayer/mplayer.conf") < 0)
   exit(1);
 if ((conffile = get_path("")) == NULL) {
   mp_msg(MSGT_CPLAYER,MSGL_WARN,MSGTR_NoHomeDir);
@@ -533,13 +533,15 @@
 
 // check codec.conf
 if(!parse_codec_cfg(get_path("codecs.conf"))){
-  if(!parse_codec_cfg(DATADIR"/codecs.conf")){
-    mp_msg(MSGT_CPLAYER,MSGL_HINT,MSGTR_CopyCodecsConf);
-//    printf("Exit.\n");
-//    exit(0);  // From unknown reason a hangup occurs here :((((((
-    kill(getpid(),SIGTERM);
-    usleep(20000);
-    kill(getpid(),SIGKILL);
+  if(!parse_codec_cfg("/etc/mplayer/codecs.conf")){
+    if(!parse_codec_cfg(DATADIR"/codecs.conf")){
+      mp_msg(MSGT_CPLAYER,MSGL_HINT,MSGTR_CopyCodecsConf);
+//      printf("Exit.\n");
+//      exit(0);  // From unknown reason a hangup occurs here :((((((
+      kill(getpid(),SIGTERM);
+      usleep(20000);
+      kill(getpid(),SIGKILL);
+    }
   }
 }
 
