--- mplayer-r32669/configure.orig	2010-11-03 09:12:22.000000000 +0100
+++ mplayer-r32669/configure	2010-11-30 09:21:10.000406064 +0100
@@ -523,7 +523,8 @@
 
 Use these options if autodetection fails:
   --extra-cflags=FLAGS        extra CFLAGS
-  --extra-ldflags=FLAGS       extra LDFLAGS
+  --real-ldflags=FLAGS        LDFLAGS
+  --extra-ldflags=FLAGS       extra LIBS !
   --extra-libs=FLAGS          extra linker flags
   --extra-libs-mplayer=FLAGS  extra linker flags for MPlayer
   --extra-libs-mencoder=FLAGS extra linker flags for MEncoder
@@ -853,6 +854,9 @@
   --extra-cflags=*)
     extra_cflags=$(echo $ac_option | cut -d '=' -f 2-)
     ;;
+  --real-ldflags=*)
+    real_ldflags=$(echo $ac_option | cut -d '=' -f 2-)
+    ;;
   --extra-ldflags=*)
     extra_ldflags=$(echo $ac_option | cut -d '=' -f 2-)
     ;;
@@ -7802,7 +7806,7 @@
 
 echocheck "compiler support for noexecstack"
 if cflag_check -Wl,-z,noexecstack ; then
-  extra_ldflags="-Wl,-z,noexecstack $extra_ldflags"
+  real_ldflags="-Wl,-z,noexecstack $real_ldflags"
   echores "yes"
 else
   echores "no"
@@ -8023,6 +8027,7 @@
 CFLAGS_SVGALIB_HELPER    = $cflags_svgalib_helper
 CFLAGS_TREMOR_LOW        = $cflags_tremor_low
 
+LDFLAGS            = $real_ldflags
 EXTRALIBS          = $extra_ldflags $_ld_static $_ld_lm $extra_libs
 EXTRALIBS_MPLAYER  = $libs_mplayer
 EXTRALIBS_MENCODER = $libs_mencoder
--- mplayer-r32669/Makefile.orig	2010-10-27 19:55:45.000000000 +0200
+++ mplayer-r32669/Makefile	2010-11-30 09:24:31.973414915 +0100
@@ -853,7 +853,7 @@
 mplayer$(EXESUF): $(MPLAYER_DEPS)
 mplayer$(EXESUF): EXTRALIBS += $(EXTRALIBS_MPLAYER)
 mencoder$(EXESUF) mplayer$(EXESUF):
-	$(CC) -o $@ $^ $(EXTRALIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(EXTRALIBS)
 
 codec-cfg$(EXESUF): codec-cfg.c codec-cfg.h help_mp.h
 	$(HOST_CC) -O -DCODECS2HTML -I. -Iffmpeg -o $@ $<
@@ -1072,7 +1072,7 @@
 TOOLS/netstream$(EXESUF): TOOLS/netstream.c
 TOOLS/vivodump$(EXESUF): TOOLS/vivodump.c
 TOOLS/netstream$(EXESUF) TOOLS/vivodump$(EXESUF): $(subst mplayer.o,mplayer-nomain.o,$(OBJS_MPLAYER)) $(filter-out %mencoder.o,$(OBJS_MENCODER)) $(OBJS_COMMON) $(COMMON_LIBS)
-	$(CC) $(CC_DEPFLAGS) $(CFLAGS) -o $@ $^ $(EXTRALIBS_MPLAYER) $(EXTRALIBS_MENCODER) $(EXTRALIBS)
+	$(CC) $(CC_DEPFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(EXTRALIBS_MPLAYER) $(EXTRALIBS_MENCODER) $(EXTRALIBS)
 
 REAL_SRCS    = $(wildcard TOOLS/realcodecs/*.c)
 REAL_TARGETS = $(REAL_SRCS:.c=.so.6.0)
