--- MPlayer-1.5/configure.orig	2023-04-11 20:43:22.582368374 +0200
+++ MPlayer-1.5/configure	2023-04-11 20:44:20.978718680 +0200
@@ -4193,6 +4193,21 @@ fi
 echores "$_alloca"
 
 
+echocheck "byteswap.h"
+cat > $TMPC << EOF
+#include <byteswap.h>
+int main(void) { (void) bswap_16(0); (void) bswap_32(0); (void) bswap_64(0); return 0; }
+EOF
+_byteswap=no
+cc_check && _byteswap=yes
+if cc_check ; then
+  def_byteswap_h='#define HAVE_BYTESWAP_H 1'
+else
+  def_byteswap_h='#undef HAVE_BYTESWAP_H'
+fi
+echores "$_byteswap"
+
+
 echocheck "fastmemcpy"
 if test "$_fastmemcpy" = yes ; then
   def_fastmemcpy='#define CONFIG_FASTMEMCPY 1'
@@ -9164,6 +9179,7 @@ $def_sys_uio_h
 /* system headers */
 $def_alloca_h
 $def_altivec_h
+$def_byteswap_h
 $def_malloc_h
 $def_mman_h
 $def_mman_has_map_failed
