--- MPlayer-0.60/configure.orig	Wed Jan  2 20:05:02 2002
+++ MPlayer-0.60/configure	Thu Jan 17 22:57:37 2002
@@ -439,7 +439,7 @@
   if test -z "$pparam" ; then
     pparam=`$_cpuinfo | grep 'flags' | cut -d ':' -f 2 | head -1`
   fi
-
+  pparam=""
   _mmx=no
   _3dnow=no
   _3dnowex=no
@@ -1268,11 +1268,11 @@
 int main(void) { return 0; }
 EOF
   _termcap=no
-  cc_check -ltermcap && _termcap=yes
+  cc_check -ltinfo && _termcap=yes
 fi
 if test "$_termcap" = yes ; then
   _def_termcap='#define USE_TERMCAP 1'
-  _ld_termcap='-ltermcap'
+  _ld_termcap='-ltinfo'
 else
   _def_termcap='#undef USE_TERMCAP'
 fi
@@ -2741,7 +2741,7 @@
 AR = ar
 CC = $_cc
 # OPTFLAGS = -O4 $_profile $_debug $_march $_mcpu -pipe -fomit-frame-pointer -ffast-math
-OPTFLAGS = $CFLAGS
+OPTFLAGS = $CFLAGS -pipe -ffast-math
 EXTRA_INC = $_inc_extra $_inc_gtk
 WIN32_PATH = -DWIN32_PATH=\\"$_win32libdir\\"
 
