--- MPlayer-1.0pre2/configure.orig	2003-11-23 00:09:38.000000000 +0100
+++ MPlayer-1.0pre2/configure	2003-11-23 00:20:52.337674440 +0100
@@ -1173,6 +1173,7 @@
 _gethostbyname2=auto
 _ftp=yes
 _vstream=auto
+_disable_altivec=no
 for ac_option do
   case "$ac_option" in
   # Skip 1st pass
@@ -1483,7 +1484,7 @@
   --enable-3dnowex) _3dnow=yes _3dnowex=yes ;;
   --disable-3dnowex) _3dnowex=no ;;
   --enable-altivec) _altivec=yes ;;
-  --disable-altivec) _altivec=no ;;
+  --disable-altivec) _altivec=no; _disable_altivec=yes ;;
   --enable-mmx)	_mmx=yes ;;
   --disable-mmx) # 3Dnow! and MMX2 require MMX
         _3dnow=no _3dnowex=no _mmx=no _mmx2=no ;;
@@ -1623,7 +1624,7 @@
 
     _altivec_gcc_flags=''
 
-    if test "$_altivec" = yes -o "$_runtime_cpudetection" = yes ; then
+    if test "$_altivec" = yes -o "$_runtime_cpudetection" = yes -a "$_disable_altivec" != "yes"; then
         echocheck "GCC altivec support"
         
         p=''
