--- MPlayer-1.0rc1/libavcodec/Makefile.orig	2006-10-23 00:32:34.000000000 +0200
+++ MPlayer-1.0rc1/libavcodec/Makefile	2006-11-17 19:02:18.581014500 +0100
@@ -265,33 +265,6 @@
 OBJS-$(CONFIG_DTS)                     += dtsdec.o
 
 OBJS-$(CONFIG_AMR)                     += amr.o
-OBJS-$(CONFIG_AMR_NB)                  += amr_float/sp_dec.o     \
-                                          amr_float/sp_enc.o     \
-                                          amr_float/interf_dec.o \
-                                          amr_float/interf_enc.o
-
-ifeq ($(CONFIG_AMR_NB_FIXED),yes)
-EXTRAOBJS += amr/*.o
-EXTRADEPS=amrlibs
-endif
-
-OBJS-$(CONFIG_AMR_WB)                  += amrwb_float/dec_acelp.o \
-                                          amrwb_float/dec_dtx.o   \
-                                          amrwb_float/dec_gain.o  \
-                                          amrwb_float/dec_if.o    \
-                                          amrwb_float/dec_lpc.o   \
-                                          amrwb_float/dec_main.o  \
-                                          amrwb_float/dec_rom.o   \
-                                          amrwb_float/dec_util.o  \
-                                          amrwb_float/enc_acelp.o \
-                                          amrwb_float/enc_dtx.o   \
-                                          amrwb_float/enc_gain.o  \
-                                          amrwb_float/enc_if.o    \
-                                          amrwb_float/enc_lpc.o   \
-                                          amrwb_float/enc_main.o  \
-                                          amrwb_float/enc_rom.o   \
-                                          amrwb_float/enc_util.o  \
-                                          amrwb_float/if_rom.o
 
 OBJS-$(CONFIG_AAC_PARSER)              += parser.o
 OBJS-$(CONFIG_AC3_PARSER)              += parser.o
@@ -411,9 +384,6 @@
 
 include $(SRC_PATH)/common.mak
 
-amrlibs:
-	$(MAKE) -C amr spclib fipoplib
-
 tests: apiexample $(TESTS)
 
 clean::
@@ -427,10 +397,7 @@
 	   sh4/*.o sh4/*~ \
 	   sparc/*.o sparc/*~ \
 	   liba52/*.o liba52/*~ \
-	   amr_float/*.o \
 	   apiexample $(TESTS)
-	-$(MAKE) -C amr clean
-	-$(MAKE) -C amrwb_float -f makefile.gcc clean
 
 apiexample: apiexample.o $(LIB)
 
--- MPlayer-1.0pre8/libavcodec/amr.c.orig	2006-06-11 20:35:48.000000000 +0200
+++ MPlayer-1.0pre8/libavcodec/amr.c	2006-07-02 21:00:37.348070750 +0200
@@ -59,16 +59,16 @@
 
 #define MMS_IO
 
-#include "amr/sp_dec.h"
-#include "amr/d_homing.h"
-#include "amr/typedef.h"
-#include "amr/sp_enc.h"
-#include "amr/sid_sync.h"
-#include "amr/e_homing.h"
+#include <amr/sp_dec.h>
+#include <amr/d_homing.h>
+#include <amr/typedef.h>
+#include <amr/sp_enc.h>
+#include <amr/sid_sync.h>
+#include <amr/e_homing.h>
 
 #else
-#include "amr_float/interf_dec.h"
-#include "amr_float/interf_enc.h"
+#include <amrnb/interf_dec.h>
+#include <amrnb/interf_enc.h>
 #endif
 
 /* Common code for fixed and float version*/
@@ -493,8 +493,8 @@
 #define typedef_h
 #endif
 
-#include "amrwb_float/enc_if.h"
-#include "amrwb_float/dec_if.h"
+#include <amrwb/enc_if.h>
+#include <amrwb/dec_if.h>
 
 /* Common code for fixed and float version*/
 typedef struct AMRWB_bitrates
--- MPlayer-1.0rc1/configure.orig	2006-11-17 18:58:32.998916500 +0100
+++ MPlayer-1.0rc1/configure	2006-11-17 19:04:19.384564250 +0100
@@ -6349,7 +6349,7 @@
 echocheck "AMR narrowband"
 if test "$_amr_nb" = auto ; then
   _amr_nb=no
-  if test -f libavcodec/amr_float/sp_dec.c ; then
+  if test -f /usr/include/amrnb/interf_dec.h ; then
     if test "$_libavcodec" = yes ; then
       _amr_nb=yes
     else
@@ -6361,6 +6361,7 @@
   _amr=yes
   _def_amr='#define CONFIG_AMR 1'
   _def_amr_nb='#define CONFIG_AMR_NB 1'
+  _ld_amd_nb='-lamrnb'
 else
   _def_amr_nb='#undef CONFIG_AMR_NB'
 fi
@@ -6369,7 +6370,7 @@
 echocheck "AMR narrowband, fixed point"
 if test "$_amr_nb_fixed" = auto ; then
   _amr_nb_fixed=no
-  if test -f libavcodec/amr/dtx_dec.c ; then
+  if test -f /usr/include/amr/sp_dec.h ; then
     if test "$_libavcodec" = yes ; then
       if test "$_amr_nb" = no ; then
         _amr_nb_fixed=yes
@@ -6385,6 +6386,7 @@
   _amr=yes
   _def_amr='#define CONFIG_AMR 1'
   _def_amr_nb_fixed='#define CONFIG_AMR_NB_FIXED 1'
+  _ld_amr_nb='-lamrnbfixed'
 else
   _def_amr_nb_fixed='#undef CONFIG_AMR_NB_FIXED'
 fi
@@ -6399,7 +6401,7 @@
 echocheck "AMR wideband"
 if test "$_amr_wb" = auto ; then
   _amr_wb=no
-  if test -f libavcodec/amrwb_float/dec_dtx.c ; then
+  if test -f /usr/include/amrwb/dec_if.h ; then
     if test "$_libavcodec" = yes ; then
       _amr_wb=yes
     else
@@ -6411,6 +6413,7 @@
   _amr=yes
   _def_amr='#define CONFIG_AMR 1'
   _def_amr_wb='#define CONFIG_AMR_WB 1'
+  _ld_amr_wb='-lamrwb'
   _codecmodules="amr_wb $_codecmodules"
 else
   _def_amr_wb='#undef CONFIG_AMR_WB'
@@ -7630,7 +7633,9 @@
 CONFIG_AMR=$_amr
 CONFIG_AMR_NB=$_amr_nb
 CONFIG_AMR_NB_FIXED=$_amr_nb_fixed
+AMR_NB_LIB=$_ld_amr_nb
 CONFIG_AMR_WB=$_amr_wb
+AMR_WB_LIB=$_ld_amr_wb
 `echo $_libavdecoders | tr '[a-z] ' '[A-Z]\n' | sed 's/^/CONFIG_/;s/$/=yes/'`
 `echo $_libavencoders | tr '[a-z] ' '[A-Z]\n' | sed 's/^/CONFIG_/;s/$/=yes/'`
 `echo $_libavparsers | tr '[a-z] ' '[A-Z]\n' | sed 's/^/CONFIG_/;s/$/=yes/'`
--- MPlayer-1.0rc1/Makefile.orig	2006-10-23 00:32:31.000000000 +0200
+++ MPlayer-1.0rc1/Makefile	2006-11-17 19:04:54.290745750 +0100
@@ -110,6 +110,8 @@
              $(MUSEPACK_LIB) \
              $(SPEEX_LIB) \
              $(NUT_LIB) \
+             $(AMR_NB_LIB) \
+             $(AMR_WB_LIB)
 
 ifeq ($(LAVC_MP3LAME),yes)
 CODEC_LIBS += $(MP3LAME_LIB)
