--- MPlayer-1.0pre8/configure.orig	2006-06-11 20:35:47.000000000 +0200
+++ MPlayer-1.0pre8/configure	2006-07-02 16:05:03.159753250 +0200
@@ -2652,7 +2652,7 @@
 #include <sys/socket.h>
 int main(void) { (void) gethostbyname(0); (void) socket(AF_INET, SOCK_STREAM, 0); return 0; }
 EOF
-for _ld_tmp in "-lsocket -lbind" "-lsocket -ldnet" "-lsocket -lnsl" "-lnsl" "-lsocket" ; do
+for _ld_tmp in "" "-lsocket -lbind" "-lsocket -ldnet" "-lsocket -lnsl" "-lnsl" "-lsocket" ; do
   cc_check $_ld_tmp && _ld_sock="$_ld_tmp" && break
 done
 if test $_winsock2 = auto && not cygwin ; then
