--- MPlayer-1.5/configure.orig	2023-04-11 21:32:54.206269705 +0200
+++ MPlayer-1.5/configure	2023-04-12 06:22:00.634312397 +0200
@@ -7374,6 +7374,8 @@ echores "$_nemesi"
 echocheck "LIVE555 Streaming Media libraries"
 if test "$_live" = yes && test "$networking" = yes; then
   def_live='#define CONFIG_LIVE555 1'
+  extra_ldflags="$extra_ldflags -lliveMedia -lgroupsock -lBasicUsageEnvironment -lUsageEnvironment -lstdc++"
+  extra_cxxflags="$extra_cxxflags -I/usr/include/liveMedia -I/usr/include/UsageEnvironment -I/usr/include/BasicUsageEnvironment -I/usr/include/groupsock"
  inputmodules="live555 $inputmodules"
 else
   _live=no
