diff -urN main.org/configure main/configure
--- main.org/configure	Fri Apr 20 00:18:07 2001
+++ main/configure	Fri Apr 20 00:18:46 2001
@@ -403,7 +403,7 @@
 $_cc $TMPC -o $TMPO -L/usr/local/lib/ -lSDL -lpthread &> /dev/null && _have_sdl=yes
 
 _termcap=no
-$_cc $TMPC -o $TMPO -ltermcap &> /dev/null && _termcap=yes
+$_cc $TMPC -o $TMPO -lncurses &> /dev/null && _termcap=yes
 
 _png=no
 $_cc $TMPC -o $TMPO -lpng &> /dev/null && _png=yes
@@ -420,7 +420,6 @@
 
 if [ $_x11 = yes ]; then
 
-$_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lXdpms &> /dev/null && _xdpms=yes
 $_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lXv &> /dev/null && _xv=yes
 
 $_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lGL &> /dev/null && _gl=yes
@@ -710,7 +709,7 @@
 
 if [ "$_termcap" = "yes" ]; then
  _termcap='#define USE_TERMCAP'
- _libtermcap='-ltermcap'
+ _libtermcap='-lncurses'
 else
  _termcap='#undef USE_TERMCAP'
  _libtermcap=''
@@ -745,7 +744,7 @@
 CC=$_cc
 X11DIR=$_x11libdir
 # OPTFLAGS=-O4 -march=$proc -mcpu=$proc -pipe -fomit-frame-pointer -ffast-math
-OPTFLAGS=-O4 -march=$proc -mcpu=$proc -pipe -ffast-math
+OPTFLAGS=$CFLAGS -pipe -ffast-math
 # LIBS=-L/usr/lib -L/usr/local/lib $_x11libdir $_gllib $_sdllib $_dgalib $_x11lib $_xvlib
 X_LIBS=$_x11libdir $_gllib $_sdllib $_dgalib $_x11lib $_xvlib $_vmlib $_svgalib $_libpng
 TERMCAP_LIB=$_libtermcap
@@ -753,7 +752,7 @@
 LIRC_LIBS = $_lirclibs
 CSS_LIB = $_csslib
 CSS_INC = $_cssinc
-WIN32_PATH=-DWIN32_PATH=\"$_win32libdir\"
+WIN32_PATH=-DWIN32_PATH=\\"$_win32libdir\\"
 
 EOF
 # echo 'CFLAGS=$(OPTFLAGS) -Wall -DMPG12PLAY' >> config.mak
diff -urN main.org/libvo/Makefile main/libvo/Makefile
--- main.org/libvo/Makefile	Fri Apr 20 00:18:08 2001
+++ main/libvo/Makefile	Fri Apr 20 00:18:21 2001
@@ -6,7 +6,7 @@
 SRCS=osd.c font_load.c rgb15to16mmx.c yuv2rgb_mmx.c yuv2rgb.c video_out.c vo_null.c vo_pgm.c vo_md5.c vo_odivx.c x11_common.c $(OPTIONAL_SRCS)
 OBJS=osd.o font_load.o rgb15to16mmx.o yuv2rgb_mmx.o yuv2rgb.o video_out.o vo_null.o vo_pgm.o vo_md5.o vo_odivx.o x11_common.o $(OPTIONAL_OBJS)
 
-CFLAGS  = $(OPTFLAGS) -I. -I.. -DMPG12PLAY
+CFLAGS  = $(OPTFLAGS) -I. -I.. -DMPG12PLAY -I/usr/X11R6/include
 # -I/usr/X11R6/include/
 
 .SUFFIXES: .c .o
diff -urN main.org/mplayer.c main/mplayer.c
--- main.org/mplayer.c	Fri Apr 20 00:18:07 2001
+++ main/mplayer.c	Fri Apr 20 00:18:21 2001
@@ -504,7 +504,7 @@
 
   printf("%s",banner_text);
 
-if (parse_config_file(conf, "/etc/mplayer.conf") < 0)
+if (parse_config_file(conf, "/etc/mplayer/mplayer.conf") < 0)
   exit(1);
 if ((conffile = get_path("")) == NULL) {
   printf("Can't find HOME dir\n");
@@ -569,8 +569,10 @@
 
 // check codec.conf
 if(!parse_codec_cfg(get_path("codecs.conf"))){
+    if (!parse_codec_cfg("/etc/mplayer/codecs.conf")) {
     printf("(copy/link DOCS/codecs.conf to ~/.mplayer/codecs.conf)\n");
     exit(1);
+    }
 }
 
 // check font
diff -urN main.org/mplayerHQ.c main/mplayerHQ.c
--- main.org/mplayerHQ.c	Fri Apr 20 00:18:07 2001
+++ main/mplayerHQ.c	Fri Apr 20 00:18:21 2001
@@ -381,7 +381,7 @@
 
   printf("%s",banner_text);
 
-if (parse_config_file(conf, "/etc/mplayer.conf") < 0)
+if (parse_config_file(conf, "/etc/mplayer/mplayer.conf") < 0)
   exit(1);
 if ((homedir = getenv("HOME")) == NULL) {
   printf("Can't find HOME dir\n");
