--- MPlayer-1.0pre7/configure.orig	2005-06-20 21:29:14.000000000 +0200
+++ MPlayer-1.0pre7/configure	2005-06-20 21:30:30.000000000 +0200
@@ -472,7 +472,7 @@
       i[3-9]86*|x86|x86pc|k5|k6|k6_2|k6_3|k6-2|k6-3|pentium*|athlon*|i586_i686|i586-i686|BePC) host_arch=i386 ;;
       ia64) host_arch=ia64 ;;
       x86_64|amd64)
-        if [ "`$_cc -dumpmachine | grep x86_64 | cut -d- -f1`" = "x86_64" -a \
+        if [ -n "`$_cc -dumpmachine | grep -E '^(x86_|amd)64-'`" -a \
              -z "`echo $CFLAGS | grep -- -m32`"  ]; then
           host_arch=x86_64
         else
