--- mplayer-1.0_rc2_p27725/configure~	2008-11-23 19:53:55.000000000 +0200
+++ mplayer-1.0_rc2_p27725/configure	2008-11-23 20:03:15.844840283 +0200
@@ -1173,6 +1173,7 @@
 _vstream=auto
 _pthreads=yes
 _rpath=no
+_disable_altivec=no
 _asmalign_pot=auto
 for ac_option do
   case "$ac_option" in
@@ -1483,7 +1484,7 @@
   --enable-3dnowex) _3dnow=yes _3dnowex=yes ;;
   --disable-3dnowex) _3dnowex=no ;;
   --enable-altivec) _altivec=yes ;;
-  --disable-altivec) _altivec=no ;;
+  --disable-altivec) _altivec=no; _disable_altivec=yes ;;
   --enable-armv5te) _armv5te=yes ;;
   --disable-armv5te) _armv5te=no ;;
   --enable-iwmmxt) _iwmmxt=yes ;;
@@ -2391,7 +2391,7 @@
 
 
 #FIXME: This should happen before the check for CFLAGS..
-if ppc && ( test "$_altivec" = yes || test "$_runtime_cpudetection" = yes ) ; then
+if ppc && ( test "$_altivec" = yes || test "$_runtime_cpudetection" = yes) && test "$_disable_altivec" != "yes"; then
 
     # check if AltiVec is supported by the compiler, and how to enable it
     echocheck "GCC AltiVec flags"
