diff -Nur MPlayer-0.50.orig/configure MPlayer-0.50/configure
--- MPlayer-0.50.orig/configure	Fri Oct 12 14:08:53 2001
+++ MPlayer-0.50/configure	Fri Oct 12 14:18:10 2001
@@ -359,10 +359,10 @@
 fi
 
 if [ "$_x11incdir" = "" ]; then
-   if [ -d /usr/include/X11 ]; then
-      _x11incdir=
-   elif [ -d /usr/X11R6 ]; then
+   if [ -d /usr/X11R6 ]; then
       _x11incdir=-I/usr/X11R6/include
+   elif [ -d /usr/include/X11 ]; then
+      _x11incdir=
    elif [ -d /usr/X11 ]; then
       _x11incdir=-I/usr/X11/include
    fi
@@ -876,7 +876,6 @@
 # for Solaris (socket stuff is in -lsocket, gethostbyname and friends in -lnsl):
 _socklib=
 $_cc $TMPC -o $TMPO $_socklib -lsocket >/dev/null 2>&1 && _socklib="$_socklib -lsocket"
-$_cc $TMPC -o $TMPO $_socklib -lnsl >/dev/null 2>&1 && _socklib="$_socklib -lnsl"
 
 if [ $_x11 = auto ]; then
   _x11=no
@@ -887,6 +886,7 @@
 
 $_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lXdpms $_socklib > /dev/null 2>&1 && _xdpms_3=yes
 nm `echo $_x11libdir|cut -c 3-`/libXext.a | grep DPMSQueryExtension > /dev/null 2>&1 && _xdpms_4=yes
+objdump -T `echo $_x11libdir|cut -c 3-`/libXext.so | grep -q '\.text.*DPMSQueryExtension' && _xdpms_4=yes
 $_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lXv $_socklib > /dev/null 2>&1 && _xv=yes
 $_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lXxf86vm $_socklib > /dev/null 2>&1 && _vm=yes
 
@@ -906,7 +906,7 @@
 $_cc $_x11incdir $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lGL -lm -pthread $_socklib > /dev/null 2>&1 || \
  { _gl=no; echo "GL includes not found!";}
 else
-$_cc $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lGL -lm $_socklib > /dev/null 2>&1 || \
+$_cc $_x11incdir $TMPC -o $TMPO $_x11libdir -lX11 -lXext -lGL -lm $_socklib > /dev/null 2>&1 || \
  { _gl=no; echo "GL includes not found!";}
 fi
 
