--- MPlayer-1.5/configure.orig	2022-02-27 10:09:00.000000000 +0100
+++ MPlayer-1.5/configure	2023-04-11 20:38:37.187247826 +0200
@@ -613,7 +613,8 @@ Advanced options:
 
 Use these options if autodetection fails:
   --extra-cflags=FLAGS        extra CFLAGS
-  --extra-ldflags=FLAGS       extra LDFLAGS
+  --real-ldflags=FLAGS        LDFLAGS
+  --extra-ldflags=FLAGS       extra LIBS !
   --extra-libs=FLAGS          extra linker flags
   --extra-libs-mplayer=FLAGS  extra linker flags for MPlayer
   --extra-libs-mencoder=FLAGS extra linker flags for MEncoder
@@ -952,6 +953,9 @@ for ac_option do
   --extra-cflags=*)
     extra_cflags=$(option_value $ac_option)
     ;;
+  --real-ldflags=*)
+    real_ldflags=$(echo $ac_option | cut -d '=' -f 2-)
+    ;;
   --extra-ldflags=*)
     extra_ldflags=$(option_value $ac_option)
     ;;
@@ -8278,7 +8282,7 @@ echores "$_crash_debug"
 
 echocheck "compiler support for noexecstack"
 if cflag_check -Wl,-z,noexecstack ; then
-  extra_ldflags="-Wl,-z,noexecstack $extra_ldflags"
+  real_ldflags="-Wl,-z,noexecstack $real_ldflags"
   echores "yes"
 else
   echores "no"
@@ -8682,6 +8686,8 @@ CFLAGS_NO_OMIT_LEAF_FRAME_POINTER = $cfl
 CFLAGS_STACKREALIGN      = $cflags_stackrealign
 CFLAGS_SVGALIB_HELPER    = $cflags_svgalib_helper
 
+LDFLAGS            = $real_ldflags
+
 EXTRALIBS          = $($filter_flags $ld_static $extra_ldflags $timer_libs $extra_libs)
 EXTRALIBS_MPLAYER  = $($filter_flags $libs_mplayer)
 EXTRALIBS_MENCODER = $($filter_flags $libs_mencoder)
--- MPlayer-1.5/Makefile.orig	2023-04-11 20:36:49.897829062 +0200
+++ MPlayer-1.5/Makefile	2023-04-11 20:41:46.636221492 +0200
@@ -748,7 +748,7 @@ mencoder$(EXESUF): EXTRALIBS += $(EXTRAL
 mplayer$(EXESUF): $(MPLAYER_DEPS)
 mplayer$(EXESUF): EXTRALIBS += $(EXTRALIBS_MPLAYER)
 mencoder$(EXESUF) mplayer$(EXESUF):
-	$(CC) $(CC_LINK_O) $^ $(EXTRALIBS)
+	$(CC) $(LDFLAGS) $(CC_LINK_O) $^ $(EXTRALIBS)
 
 codec-cfg-test$(EXESUF): HOSTCFLAGS := $(HOSTCFLAGS) -DTESTING
 codec-cfg$(EXESUF) codecs2html$(EXESUF):  HOSTCFLAGS := $(HOSTCFLAGS) -DCODECS2HTML
@@ -790,7 +790,7 @@ version.h: version.sh $(wildcard .svn/en
 	./$< `$(CC) -dumpversion`
 
 %$(EXESUF): %.c
-	$(CC) $(CC_DEPFLAGS) $(CFLAGS) $(CC_LINK_O) $^ $(LIBS)
+	$(CC) $(LDFLAGS) $(CC_DEPFLAGS) $(CFLAGS) $(CC_LINK_O) $^ $(LIBS)
 
 %.ho: %.h
 	$(CC) $(CFLAGS) -Wno-unused -c $(CC_O) -x c $<
