--- MPlayer-1.0pre2/libmpdemux/http.c~	Tue Apr  6 16:37:57 2004
+++ MPlayer-1.0pre2/libmpdemux/http.c	Tue Apr  6 16:37:57 2004
@@ -175,7 +175,7 @@
 	if( http_hdr->method==NULL ) http_set_method( http_hdr, "GET");
 	if( http_hdr->uri==NULL ) http_set_uri( http_hdr, "/");
 	else {
-		uri = (char*)malloc(strlen(http_hdr->uri)*2);
+		uri = (char*)malloc((strlen(http_hdr->uri)*3) + 1);
 		if( uri==NULL ) {
 			mp_msg(MSGT_NETWORK,MSGL_ERR,"Memory allocation failed\n");
 			return NULL;
