--- mplayer-r29930/libdvdread4/dvdread/dvd_reader.h~	2009-11-19 14:23:23.231276231 +0100
+++ mplayer-r29930/libdvdread4/dvdread/dvd_reader.h	2009-11-19 14:25:48.524614054 +0100
@@ -74,6 +74,8 @@
  */
 typedef struct dvd_stat_s dvd_stat_t;
 
+#pragma GCC visibility push(default)
+
 /**
  * Opens a block device of a DVD-ROM file, or an image file, or a directory
  * name for a mounted DVD or HD copy of a DVD.
@@ -305,6 +307,8 @@
  */
 int DVDUDFCacheLevel( dvd_reader_t *, int );
 
+#pragma GCC visibility pop
+
 #ifdef __cplusplus
 };
 #endif
--- mplayer-r29930/libdvdread4/dvdread/ifo_read.h~	2009-11-19 14:23:00.545070443 +0100
+++ mplayer-r29930/libdvdread4/dvdread/ifo_read.h	2009-11-19 14:26:19.125441477 +0100
@@ -29,6 +29,7 @@
 extern "C" {
 #endif
 
+#pragma GCC visibility push(default)
 /**
  * handle = ifoOpen(dvd, title);
  *
@@ -226,6 +227,8 @@
 void ifoFree_TITLE_VOBU_ADMAP(ifo_handle_t *);
 void ifoFree_TXTDT_MGI(ifo_handle_t *);
 
+#pragma GCC visibility pop
+
 #ifdef __cplusplus
 };
 #endif
--- mplayer-r29930/libdvdread4/dvdread/nav_read.h~	2009-11-18 00:00:29.000000000 +0100
+++ mplayer-r29930/libdvdread4/dvdread/nav_read.h	2009-11-19 14:27:13.724600811 +0100
@@ -31,6 +31,7 @@
 extern "C" {
 #endif
 
+#pragma GCC visibility push(default)
 /**
  * Reads the PCI packet data pointed to into th pci struct.
  *
@@ -46,6 +47,7 @@
  * @param bufffer Pointer to the buffer of the on disc DSI data.
  */
 void navRead_DSI(dsi_t *, unsigned char *);
+#pragma GCC visibility pop
 
 #ifdef __cplusplus
 };
--- MPlayer-1.1/configure.orig	2013-02-24 17:53:33.092693098 +0100
+++ MPlayer-1.1/configure	2013-02-24 19:29:59.079116950 +0100
@@ -7426,6 +7426,7 @@
   _vstream=no
   cat > $TMPC <<EOF
 #include <vstream-client.h>
+#pragma GCC visibility push(default)
 void vstream_error(const char *format, ... ) {}
 int main(void) { vstream_start(); return 0; }
 EOF
--- MPlayer-1.1/stream/stream_vstream.c.orig	2011-10-21 17:44:58.000000000 +0200
+++ MPlayer-1.1/stream/stream_vstream.c	2013-02-24 21:03:09.752215665 +0100
@@ -51,6 +51,7 @@
 
 #include <vstream-client.h>
 
+#pragma GCC visibility push(default)
 void vstream_error(const char *format, ...) {
     char buf[1024];
     va_list va;
@@ -59,6 +60,7 @@
     va_end(va);
     mp_msg(MSGT_STREAM, MSGL_ERR, buf);
 }
+#pragma GCC visibility pop
 
 static struct stream_priv_s {
   char* host;
