--- MPlayer-0.60/configure.orig	Wed Jan  2 20:05:02 2002
+++ MPlayer-0.60/configure	Thu Jan 17 22:57:37 2002
@@ -1640,11 +1640,11 @@
 int main(void) { return 0; }
 EOF
   _termcap=no
-  cc_check -ltermcap && _termcap=yes
+  cc_check -ltinfo && _termcap=yes
 fi
 if test "$_termcap" = yes ; then
   _def_termcap='#define USE_TERMCAP 1'
-  _ld_termcap='-ltermcap'
+  _ld_termcap='-ltinfo'
 else
   _def_termcap='#undef USE_TERMCAP'
 fi
