diff -durN MPlayer-0.18pre5.orig/mplayer.c MPlayer-0.18pre5/mplayer.c
--- MPlayer-0.18pre5.orig/mplayer.c	Mon Jul  2 21:25:34 2001
+++ MPlayer-0.18pre5/mplayer.c	Fri Jul  6 11:40:53 2001
@@ -466,7 +466,7 @@
 {
 char *conffile;
 int conffile_fd;
-if (parse_config_file(conf, "/etc/mplayer.conf") < 0)
+if (parse_config_file(conf, "/etc/mplayer/mplayer.conf") < 0)
   exit(1);
 if ((conffile = get_path("")) == NULL) {
   printf("Can't find HOME dir\n");
@@ -631,9 +631,11 @@
 
 // check codec.conf
 if(!parse_codec_cfg(get_path("codecs.conf"))){
+    if (!parse_codec_cfg("/etc/mplayer/codecs.conf")) {
     fprintf(stderr,"(copy/link DOCS/codecs.conf to ~/.mplayer/codecs.conf)\n");
     GUI_MSG( mplCodecConfNotFound )
     exit(1);
+    }
 }
 
 // check font
