--- mplayer-export-2014-09-12/libvo/vo_gif89a.c.orig	2014-01-18 22:39:07.000000000 +0100
+++ mplayer-export-2014-09-12/libvo/vo_gif89a.c	2014-09-16 17:01:13.843265608 +0200
@@ -367,7 +367,11 @@
 			MPLAYER_VERSION, VO_GIF_REVISION,
 			"joey@nicewarrior.org");
 		EGifPutComment(new_gif, temp);
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+		EGifCloseFile(new_gif, NULL); // also frees gif storage space.
+#else
 		EGifCloseFile(new_gif); // also frees gif storage space.
+#endif
 	}
 
 	// free our allocated ram
--- mplayer-export-2014-09-12/libmpdemux/demux_gif.c.orig	2014-02-23 19:14:44.000000000 +0100
+++ mplayer-export-2014-09-12/libmpdemux/demux_gif.c	2014-09-16 18:00:40.546449439 +0200
@@ -51,6 +51,9 @@
 #define GifError() (gif ? gif->Error : 0)
 #define GifErrorString() GifErrorString(gif->Error)
 #endif
+#if defined GIFLIB_MAJOR && ((GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1)))
+#define DGifCloseFile(a) DGifCloseFile(a, NULL)
+#endif
 
 /* >= 4.2 prior GIFLIB did not have MAJOR/MINOR defines */
 #if defined GIFLIB_MAJOR && GIFLIB_MAJOR >= 4
