--- MPlayer-1.0rc1/configure.orig	2006-10-25 05:28:59.506383441 +0200
+++ MPlayer-1.0rc1/configure	2006-10-25 05:34:36.560143481 +0200
@@ -1724,6 +1724,7 @@
 _ass=auto
 _rpath=no
 _asmalign_pot=auto
+_disable_altivec=no
 for ac_option do
   case "$ac_option" in
   # Skip 1st pass
@@ -2130,7 +2131,7 @@
   --enable-3dnowext) _3dnow=yes _3dnowext=yes ;;
   --disable-3dnowext) _3dnowext=no ;;
   --enable-altivec) _altivec=yes ;;
-  --disable-altivec) _altivec=no ;;
+  --disable-altivec) _altivec=no; _disable_altivec=yes ;;
   --enable-armv5te) _armv5te=yes ;;
   --disable-armv5te) _armv5te=no ;;
   --enable-iwmmxt) _iwmmxt=yes ;;
@@ -2284,7 +2285,7 @@
 
     _altivec_gcc_flags=''
 
-    if test "$_altivec" = yes -o "$_runtime_cpudetection" = yes ; then
+    if test "$_altivec" = yes -o "$_runtime_cpudetection" = yes -a "$_disable_altivec" != "yes"; then
         echocheck "GCC altivec support"
         
         p=''
