--- MPlayer-1.5/configure.orig	2022-02-27 10:09:00.000000000 +0100
+++ MPlayer-1.5/configure	2023-04-11 20:38:37.187247826 +0200
@@ -613,7 +613,8 @@ Advanced options:
 
 Use these options if autodetection fails:
   --extra-cflags=FLAGS        extra CFLAGS
-  --extra-ldflags=FLAGS       extra LDFLAGS
+  --real-ldflags=FLAGS        LDFLAGS
+  --extra-ldflags=FLAGS       extra LIBS !
   --extra-libs=FLAGS          extra linker flags
   --extra-libs-mplayer=FLAGS  extra linker flags for MPlayer
   --extra-libs-mencoder=FLAGS extra linker flags for MEncoder
@@ -952,6 +953,9 @@ for ac_option do
   --extra-cflags=*)
     extra_cflags=$(option_value $ac_option)
     ;;
+  --real-ldflags=*)
+    real_ldflags=$(echo $ac_option | cut -d '=' -f 2-)
+    ;;
   --extra-ldflags=*)
     extra_ldflags=$(option_value $ac_option)
     ;;
@@ -6486,6 +6486,13 @@
 echores "$_bluray"
 
 echocheck "dvdread"
+if test "$_dvdread" = yes ; then
+    _dvdreadcflags=$($_dvdreadconfig --cflags 2> /dev/null)
+    _dvdreadlibs=$($_dvdreadconfig --libs 2> /dev/null)
+    extra_cflags="$extra_cflags $_dvdreadcflags"
+    extra_ldflags="$extra_ldflags $_dvdreadlibs"
+fi
+echocheck "dvdread"
 if test "$_dvdread" = auto ; then
   _dvdread=no
     _dvdreadcflags=$($_dvdreadconfig --cflags 2> /dev/null)
@@ -6939,6 +6939,9 @@
   extra_ldflags="$extra_ldflags -logg -lvorbisidec"
 elif test "$_libvorbis" = yes ; then
   _vorbis=yes
+  for vorbislibs in '-lvorbisenc -lvorbis -logg' '-lvorbis -logg' ; do
+    statement_check vorbis/vorbisenc.h 'vorbis_encode_ctl(0, 0, 0); ogg_stream_clear(0)' $vorbislibs && _libvorbis=yes && break
+  done
   def_vorbis='#define CONFIG_OGGVORBIS 1'
   codecmodules="libvorbis $codecmodules"
   res_comment="libvorbis"
@@ -7023,11 +7023,12 @@
 if test "$_mpg123" = auto; then
   _mpg123=no
   statement_check mpg123.h 'mpg123_param(NULL, MPG123_RESYNC_LIMIT, -1, 0.)' -lmpg123 &&
-      _mpg123=yes && extra_ldflags="$extra_ldflags -lmpg123"
+      _mpg123=yes
 fi
 if test "$_mpg123" = yes ; then
   def_mpg123='#define CONFIG_MPG123 1'
   codecmodules="mpg123 $codecmodules"
+  extra_ldflags="$extra_ldflags -lmpg123"
 else
   nocodecmodules="mpg123 $nocodecmodules"
 fi
@@ -7038,11 +7039,12 @@
 if test "$_liba52" = auto ; then
   _liba52=no
   statement_check_broken inttypes.h a52dec/a52.h 'a52_state_t *testHand; testHand=a52_init(0)' -la52 &&
-    _liba52=yes && extra_ldflags="$extra_ldflags -la52"
+    _liba52=yes
 fi
 if test "$_liba52" = yes ; then
   def_liba52='#define CONFIG_LIBA52 1'
   codecmodules="liba52 $codecmodules"
+  extra_ldflags="$extra_ldflags -la52"
 else
   nocodecmodules="liba52 $nocodecmodules"
 fi
@@ -7066,7 +7068,10 @@
 if test "$_libmpeg2" = auto ; then
   _libmpeg2=no
   header_check_broken stdint.h mpeg2dec/mpeg2.h -lmpeg2 &&
-    _libmpeg2=yes && extra_ldflags="$extra_ldflags -lmpeg2"
+    _libmpeg2=yes
+fi
+if test "$_libmpeg2" = yes ; then
+    extra_ldflags="$extra_ldflags -lmpeg2"
 fi
 
 def_libmpeg2='#undef CONFIG_LIBMPEG2'
@@ -8278,7 +8282,7 @@ echores "$_crash_debug"
 
 echocheck "compiler support for noexecstack"
 if cflag_check -Wl,-z,noexecstack ; then
-  extra_ldflags="-Wl,-z,noexecstack $extra_ldflags"
+  real_ldflags="-Wl,-z,noexecstack $real_ldflags"
   echores "yes"
 else
   echores "no"
@@ -8682,6 +8686,8 @@ CFLAGS_NO_OMIT_LEAF_FRAME_POINTER = $cfl
 CFLAGS_STACKREALIGN      = $cflags_stackrealign
 CFLAGS_SVGALIB_HELPER    = $cflags_svgalib_helper
 
+LDFLAGS            = $real_ldflags
+
 EXTRALIBS          = $($filter_flags $ld_static $extra_ldflags $timer_libs $extra_libs)
 EXTRALIBS_MPLAYER  = $($filter_flags $libs_mplayer)
 EXTRALIBS_MENCODER = $($filter_flags $libs_mencoder)
--- MPlayer-1.5/Makefile.orig	2023-04-11 20:36:49.897829062 +0200
+++ MPlayer-1.5/Makefile	2023-04-11 20:41:46.636221492 +0200
@@ -748,7 +748,7 @@ mencoder$(EXESUF): EXTRALIBS += $(EXTRAL
 mplayer$(EXESUF): $(MPLAYER_DEPS)
 mplayer$(EXESUF): EXTRALIBS += $(EXTRALIBS_MPLAYER)
 mencoder$(EXESUF) mplayer$(EXESUF):
-	$(CC) $(CC_LINK_O) $^ $(EXTRALIBS)
+	$(CC) $(LDFLAGS) $(CC_LINK_O) $^ $(EXTRALIBS)
 
 codec-cfg-test$(EXESUF): HOSTCFLAGS := $(HOSTCFLAGS) -DTESTING
 codec-cfg$(EXESUF) codecs2html$(EXESUF):  HOSTCFLAGS := $(HOSTCFLAGS) -DCODECS2HTML
@@ -790,7 +790,7 @@ version.h: version.sh $(wildcard .svn/en
 	./$< `$(CC) -dumpversion`
 
 %$(EXESUF): %.c
-	$(CC) $(CC_DEPFLAGS) $(CFLAGS) $(CC_LINK_O) $^ $(LIBS)
+	$(CC) $(LDFLAGS) $(CC_DEPFLAGS) $(CFLAGS) $(CC_LINK_O) $^ $(LIBS)
 
 %.ho: %.h
 	$(CC) $(CFLAGS) -Wno-unused -c $(CC_O) -x c $<
