--- MPlayer-1.0pre2/libavcodec/alpha/mpegvideo_alpha.c.orig	2003-10-05 18:12:00.000000000 +0200
+++ MPlayer-1.0pre2/libavcodec/alpha/mpegvideo_alpha.c	2003-11-23 00:01:14.955663712 +0100
@@ -29,6 +29,7 @@
     uint64_t correction;
     DCTELEM *orig_block = block;
     DCTELEM block0;
+    int use_mvi;
 
     qadd = WORD_VEC((qscale - 1) | 1);
     qmul = qscale << 1;
@@ -49,6 +50,11 @@
         n_coeffs = s->intra_scantable.raster_end[s->block_last_index[n]];
     }
 
+#ifdef RUNTIME_CPUDETECT
+    use_mvi = (amask(AMASK_MVI) == 0);
+#else
+    use_mvi = 1;
+#endif
     for(i = 0; i <= n_coeffs; block += 4, i += 4) {
         uint64_t levels, negmask, zeros, add;
 
@@ -59,13 +65,18 @@
 #ifdef __alpha_max__
         /* I don't think the speed difference justifies runtime
            detection.  */
+	/* but - you cannot use MVI if building with runtime detection
+	 * and at runtime CPU doesn't support it */
+	if(use_mvi) {
         negmask = maxsw4(levels, -1); /* negative -> ffff (-1) */
         negmask = minsw4(negmask, 0); /* positive -> 0000 (0) */
-#else
+	} else
+#endif
+	{
         negmask = cmpbge(WORD_VEC(0x7fff), levels);
         negmask &= (negmask >> 1) | (1 << 7);
         negmask = zap(-1, negmask);
-#endif
+	}
 
         zeros = cmpbge(0, levels);
         zeros &= zeros >> 1;
--- MPlayer-1.0pre2/configure.orig	2003-10-05 17:34:40.000000000 +0200
+++ MPlayer-1.0pre2/configure	2003-11-23 00:09:38.111172448 +0100
@@ -996,6 +996,11 @@
   if ppc; then
     _altivec=yes
   fi
+  if test "$host_arch" = "alpha" -a "$_cc_major" -ge "3"; then
+    _def_gcc_mvi_support="#define CAN_COMPILE_ALPHA_MVI 1"
+  else
+    _def_gcc_mvi_support="#undef CAN_COMPILE_ALPHA_MVI"
+  fi
 fi
 
 if x86 && test "$_runtime_cpudetection" = no ; then
