--- MPlayer-1.5/configure.orig	2023-04-11 20:46:25.264712032 +0200
+++ MPlayer-1.5/configure	2023-04-11 20:47:41.344299874 +0200
@@ -8106,7 +8106,7 @@ fi
 if test "$_vstream" = yes ; then
   def_vstream='#define CONFIG_VSTREAM 1'
   inputmodules="vstream $inputmodules"
-  extra_ldflags="$extra_ldflags -lvstream-client"
+  ndextra_ldflags="$ndextra_ldflags -lvstream-client"
 else
   noinputmodules="vstream $noinputmodules"
   def_vstream='#undef CONFIG_VSTREAM'
@@ -8699,7 +8699,7 @@ CFLAGS_SVGALIB_HELPER    = $cflags_svgal
 
 LDFLAGS            = $real_ldflags
 
-EXTRALIBS          = $($filter_flags $ld_static $extra_ldflags $timer_libs $extra_libs)
+EXTRALIBS          = $($filter_flags $extra_ldflags $ndextra_ldflags $ld_static $timer_libs $extra_libs)
 EXTRALIBS_MPLAYER  = $($filter_flags $libs_mplayer)
 EXTRALIBS_MENCODER = $($filter_flags $libs_mencoder)
 
