--- MPlayer-1.5/mplayer.c.orig	2021-01-25 20:10:32.000000000 +0100
+++ MPlayer-1.5/mplayer.c	2025-08-14 19:41:56.394431756 +0200
@@ -2430,7 +2430,8 @@ int reinit_video_chain(void)
     return 1;
 
 err_out:
-    mpctx->sh_video = mpctx->sh_video->ds = NULL;
+    mpctx->sh_video = NULL;
+    mpctx->sh_video->ds = NULL;
     return 0;
 }
 
@@ -3527,7 +3528,7 @@ goto_enable_cache:
                 break;
             if ((mpctx->demuxer->file_format == DEMUXER_TYPE_AVI || mpctx->demuxer->file_format == DEMUXER_TYPE_ASF || mpctx->demuxer->file_format == DEMUXER_TYPE_MOV)
                 && stream_dump_type == 2)
-                stream_write_buffer(os, &in_size, 4);
+                stream_write_buffer(os, (void*)&in_size, 4);
             if (in_size > 0) {
                 stream_write_buffer(os, start, in_size);
                 stream_dump_progress(in_size, mpctx->stream);
--- MPlayer-1.5/libvo/vo_aa.c.orig	2021-01-23 19:22:09.000000000 +0100
+++ MPlayer-1.5/libvo/vo_aa.c	2025-08-14 19:47:21.889668398 +0200
@@ -357,7 +357,7 @@ draw_frame(uint8_t *src[]) {
     break;
   }
 
-  sws_scale(sws,src,stride,0,src_height,image,image_stride);
+  sws_scale(sws,(const uint8_t * const*)src,stride,0,src_height,image,image_stride);
 
    /* Now 'ASCIInate' the image */
   if (fast)
@@ -377,7 +377,7 @@ draw_slice(uint8_t *src[], int stride[],
   int dx2 = screen_x + ((x+w) * screen_w / src_width);
   int dy2 = screen_y + ((y+h) * screen_h / src_height);
 
-  sws_scale(sws,src,stride,y,h,image,image_stride);
+  sws_scale(sws,(const uint8_t * const*)src,stride,y,h,image,image_stride);
 
   /* Now 'ASCIInate' the image */
   if (fast)
--- MPlayer-1.5/libvo/vo_matrixview.c.orig	2021-01-23 19:22:09.000000000 +0100
+++ MPlayer-1.5/libvo/vo_matrixview.c	2025-08-14 19:49:36.047119628 +0200
@@ -170,7 +170,7 @@ static void flip_page(void)
 
 static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
 {
-    sws_scale(sws, src, stride, y, h, map_image, map_stride);
+    sws_scale(sws, (const uint8_t* const*) src, stride, y, h, map_image, map_stride);
     return 0;
 }
 
--- MPlayer-1.5/libvo/mga_template.c.orig	2016-02-20 14:30:50.000000000 +0100
+++ MPlayer-1.5/libvo/mga_template.c	2025-08-14 19:50:18.824036495 +0200
@@ -77,7 +77,7 @@ draw_slice_g200(uint8_t *image[], int st
 	av_image_fill_pointers(dst, AV_PIX_FMT_NV12, mga_vid_config.src_height,
 	                       vid_data, dst_stride);
 
-	sws_scale(sws_ctx, image, stride, y, height, dst, dst_stride);
+	sws_scale(sws_ctx, (const uint8_t *const *)image, stride, y, height, dst, dst_stride);
 }
 
 static void
--- MPlayer-1.5/libvo/vo_x11.c.orig	2014-01-18 22:39:07.000000000 +0100
+++ MPlayer-1.5/libvo/vo_x11.c	2025-08-14 19:50:41.150833750 +0200
@@ -492,7 +492,7 @@ static int draw_slice(uint8_t * src[], i
         dst[0] += dstStride[0] * (image_height - 1);
         dstStride[0] = -dstStride[0];
     }
-    sws_scale(swsContext, src, stride, y, h, dst, dstStride);
+    sws_scale(swsContext, (const uint8_t *const *)src, stride, y, h, dst, dstStride);
     return 0;
 }
 
--- MPlayer-1.5/mp_msg.c.orig	2021-01-23 19:22:09.000000000 +0100
+++ MPlayer-1.5/mp_msg.c	2025-08-14 19:52:44.718223158 +0200
@@ -70,7 +70,7 @@ const char* filename_recode(const char*
     filename_len = strlen(filename);
     max_path = MSGSIZE_MAX - 4;
     precoded = recoded_filename;
-    if (iconv(inv_msgiconv, &filename, &filename_len,
+    if (iconv(inv_msgiconv, (char **)&filename, &filename_len,
               &precoded, &max_path) == (size_t)(-1) && errno == E2BIG) {
         precoded[0] = precoded[1] = precoded[2] = '.';
         precoded += 3;
--- MPlayer-1.5/libmpcodecs/vf_scale.c.orig	2020-11-01 16:07:47.000000000 +0100
+++ MPlayer-1.5/libmpcodecs/vf_scale.c	2025-08-14 20:19:47.154379638 +0200
@@ -439,14 +439,14 @@ static void scale(struct SwsContext *sws
         int src_stride2[MP_MAX_PLANES]={2*src_stride[0], 2*src_stride[1], 2*src_stride[2], 2*src_stride[3]};
         int dst_stride2[MP_MAX_PLANES]={2*dst_stride[0], 2*dst_stride[1], 2*dst_stride[2], 2*dst_stride[3]};
 
-        sws_scale(sws1, src2, src_stride2, y>>1, h>>1, dst2, dst_stride2);
+        sws_scale(sws1, (const uint8_t *const *)src2, src_stride2, y>>1, h>>1, dst2, dst_stride2);
         for(i=0; i<MP_MAX_PLANES; i++){
             src2[i] += src_stride[i];
             dst2[i] += dst_stride[i];
         }
-        sws_scale(sws2, src2, src_stride2, y>>1, h>>1, dst2, dst_stride2);
+        sws_scale(sws2, (const uint8_t *const *)src2, src_stride2, y>>1, h>>1, dst2, dst_stride2);
     }else{
-        sws_scale(sws1, src2, src_stride, y, h, dst, dst_stride);
+        sws_scale(sws1, (const uint8_t *const *)src2, src_stride, y, h, dst, dst_stride);
     }
 }
 
--- MPlayer-1.5/libmpdemux/demux_film.c.orig	2021-01-23 19:22:13.000000000 +0100
+++ MPlayer-1.5/libmpdemux/demux_film.c	2025-08-14 20:21:07.488182864 +0200
@@ -157,7 +157,7 @@ static int demux_film_fill_buffer(demuxe
         free(buf);
       }
       else {/* for 16bit */
-        unsigned short* tmp = dp->buffer;
+        unsigned short* tmp = (void*)(dp->buffer);
         unsigned short* buf = malloc(film_chunk.chunk_size);
         for(i = 0; i < film_chunk.chunk_size/4; i++) {
           buf[i*2] = tmp[i];
--- MPlayer-1.5/loader/qtx/qtxsdk/components.h.orig	2025-08-14 21:14:59.937138889 +0200
+++ MPlayer-1.5/loader/qtx/qtxsdk/components.h	2025-08-14 21:15:02.530487472 +0200
@@ -749,6 +749,7 @@ static inline void dump_CodecCapabilitie
     printf("=========================================================\n");
 }
 
+#if 0
 static inline void dump_CodecDecompressParams(void* xxx){
     CodecDecompressParams* cd=xxx;
     ImageDescription **idh;
@@ -797,5 +798,6 @@ static inline void dump_CodecDecompressP
 
 
 }
+#endif
 
 #endif /* MPLAYER_COMPONENTS_H */
--- MPlayer-1.5/sub/spudec.c.orig	2021-01-23 19:22:09.000000000 +0100
+++ MPlayer-1.5/sub/spudec.c	2025-08-14 21:15:42.167387194 +0200
@@ -899,7 +899,7 @@ static void sws_spu_image(unsigned char
 	ctx=sws_getContext(sw, sh, AV_PIX_FMT_GRAY8, dw, dh, AV_PIX_FMT_GRAY8, SWS_GAUSS, &filter, NULL, NULL);
 	sws_scale(ctx,&s1,&ss,0,sh,&d1,&ds);
 	for (i=ss*sh-1; i>=0; i--) s2[i] = -s2[i];
-	sws_scale(ctx,&s2,&ss,0,sh,&d2,&ds);
+	sws_scale(ctx,(const uint8_t* const*)&s2,&ss,0,sh,&d2,&ds);
 	for (i=ds*dh-1; i>=0; i--) d2[i] = -d2[i];
 	sws_freeContext(ctx);
 }
--- MPlayer-1.5/sub/sub.c.orig	2020-06-14 19:55:21.000000000 +0200
+++ MPlayer-1.5/sub/sub.c	2025-08-14 21:16:31.554344245 +0200
@@ -727,7 +727,7 @@ static inline void vo_update_text_sub(mp
 	    // reading the subtitle words from vo_sub->text[]
           while (*t) {
             if (sub_utf8)
-              c = utf8_get_char(&t);
+              c = utf8_get_char((const char**)&t);
             else if ((c = *t++) >= 0x80 && sub_unicode)
               c = (c<<8) + *t++;
 	      if (k==MAX_UCS){
--- MPlayer-1.5/libaf/af_lavcresample.c.orig	2016-02-13 18:32:14.000000000 +0100
+++ MPlayer-1.5/libaf/af_lavcresample.c	2025-08-14 21:19:47.702164215 +0200
@@ -156,7 +156,7 @@ static af_data_t* play(struct af_instanc
 
   memcpy(s->in[0], in, in_len);
 
-  ret = swr_convert(s->swrctx, &s->tmp[0], out_len/chans/2, &s->in[0], in_len/chans/2);
+  ret = swr_convert(s->swrctx, &s->tmp[0], out_len/chans/2, (const uint8_t**)&s->in[0], in_len/chans/2);
   if (ret < 0) return NULL;
   out_len= ret*chans*2;
 
--- MPlayer-1.5/libmpcodecs/vd_ffmpeg.c.orig	2021-05-01 19:45:31.000000000 +0200
+++ MPlayer-1.5/libmpcodecs/vd_ffmpeg.c	2025-08-14 21:22:12.163013613 +0200
@@ -739,7 +739,7 @@ static int get_buffer(AVCodecContext *av
     if (ctx->use_vdpau) {
         VdpVideoSurface surface = (VdpVideoSurface)mpi->priv;
         avctx->draw_horiz_band= NULL;
-        mpi->planes[3] = surface;
+        mpi->planes[3] = (void*)surface;
     }
 #endif
 #if CONFIG_XVMC
--- MPlayer-1.5/libmpcodecs/vf_screenshot.c.orig	2021-05-01 19:45:31.000000000 +0200
+++ MPlayer-1.5/libmpcodecs/vf_screenshot.c	2025-08-14 21:23:33.090156112 +0200
@@ -60,7 +60,7 @@ static void draw_slice(struct vf_instanc
                        int* stride, int w,int h, int x, int y)
 {
     if (vf->priv->store_slices) {
-        sws_scale(vf->priv->ctx, src, stride, y, h, vf->priv->pic->data, vf->priv->pic->linesize);
+        sws_scale(vf->priv->ctx, (const uint8_t *const *)src, stride, y, h, vf->priv->pic->data, vf->priv->pic->linesize);
     }
     vf_next_draw_slice(vf,src,stride,w,h,x,y);
 }
@@ -158,7 +158,7 @@ static void scale_image(struct vf_priv_s
     if (!priv->pic->data[0])
         priv->pic->data[0] = av_malloc(priv->pic->linesize[0]*priv->dh);
 
-    sws_scale(priv->ctx, mpi->planes, mpi->stride, 0, mpi->height, priv->pic->data, priv->pic->linesize);
+    sws_scale(priv->ctx, (const uint8_t *const *)mpi->planes, mpi->stride, 0, mpi->height, priv->pic->data, priv->pic->linesize);
 }
 
 static void start_slice(struct vf_instance *vf, mp_image_t *mpi)
--- MPlayer-1.5/libmpcodecs/vf_pp.c.orig	2017-04-29 13:09:30.000000000 +0200
+++ MPlayer-1.5/libmpcodecs/vf_pp.c	2025-08-14 21:24:40.580552940 +0200
@@ -142,7 +142,7 @@ static int put_image(struct vf_instance
 
     if(vf->priv->pp || !(mpi->flags&MP_IMGFLAG_DIRECT)){
 	// do the postprocessing! (or copy if no DR)
-	pp_postprocess(mpi->planes           ,mpi->stride,
+	pp_postprocess((const uint8_t**)mpi->planes           ,mpi->stride,
 		    vf->dmpi->planes,vf->dmpi->stride,
 		    (mpi->w+7)&(~7),mpi->h,
 		    mpi->qscale, mpi->qstride,
--- MPlayer-1.5/libmpcodecs/ve_raw.c.orig	2016-02-20 12:32:40.000000000 +0100
+++ MPlayer-1.5/libmpcodecs/ve_raw.c	2025-08-14 21:40:15.189379358 +0200
@@ -150,7 +150,7 @@ static int query_format(struct vf_instan
     return 0;
 }
 
-static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts) {
+static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double) {
     mux_v->buffer = mpi->planes[0];
     muxer_write_chunk(mux_v, mpi->width*mpi->height*mux_v->bih->biBitCount/8, 0x10, pts, pts);
     return 1;
--- MPlayer-1.5/libmpdemux/muxer_avi.c.orig	2018-01-06 01:54:40.000000000 +0100
+++ MPlayer-1.5/libmpdemux/muxer_avi.c	2025-08-14 21:55:45.551513762 +0200
@@ -142,8 +142,8 @@ static muxer_stream_t* avifile_new_strea
 static void write_avi_chunk(stream_t *stream,unsigned int id,int len,void* data){
  int le_len = le2me_32(len);
  int le_id = le2me_32(id);
- stream_write_buffer(stream, &le_id, 4);
- stream_write_buffer(stream, &le_len, 4);
+ stream_write_buffer(stream, (void*)&le_id, 4);
+ stream_write_buffer(stream, (void*)&le_len, 4);
 
 if(len>0){
   if(data){
@@ -176,9 +176,9 @@ static void write_avi_list(stream_t *str
   list_id = le2me_32(list_id);
   le_len  = le2me_32(len);
   le_id   = le2me_32(id);
-  stream_write_buffer(stream, &list_id, 4);
-  stream_write_buffer(stream, &le_len, 4);
-  stream_write_buffer(stream, &le_id, 4);
+  stream_write_buffer(stream, (void*)&list_id, 4);
+  stream_write_buffer(stream, (void*)&le_len, 4);
+  stream_write_buffer(stream, (void*)&le_id, 4);
 }
 
 static void avifile_odml_new_riff(muxer_t *muxer)
@@ -199,7 +199,7 @@ static void avifile_odml_new_riff(muxer_
     riff[0]=le2me_32(mmioFOURCC('R','I','F','F'));
     riff[1]=0;
     riff[2]=le2me_32(mmioFOURCC('A','V','I','X'));
-    stream_write_buffer(muxer->stream, riff, 12);
+    stream_write_buffer(muxer->stream, (void*)riff, 12);
 
     write_avi_list(muxer->stream,listtypeAVIMOVIE,0);
 
@@ -251,12 +251,12 @@ static void avifile_write_header(muxer_t
           movilen = le2me_32(rifflen - 12);
           rifflen = le2me_32(rifflen);
           stream_seek(muxer->stream, vsi->riffofs[i]+4);
-          stream_write_buffer(muxer->stream,&rifflen,4);
+          stream_write_buffer(muxer->stream,(void*)&rifflen,4);
 
           /* fixup movi length */
           if (i > 0) {
               stream_seek(muxer->stream, vsi->riffofs[i]+16);
-              stream_write_buffer(muxer->stream,&movilen,4);
+              stream_write_buffer(muxer->stream,(void*)&movilen,4);
           }
       }
 
@@ -269,7 +269,7 @@ static void avifile_write_header(muxer_t
     riff[0]=le2me_32(riff[0]);
     riff[1]=le2me_32(riff[1]);
     riff[2]=le2me_32(riff[2]);
-    stream_write_buffer(muxer->stream,&riff,12);
+    stream_write_buffer(muxer->stream,(void*)&riff,12);
   }
 
   // update AVI header:
@@ -406,7 +406,7 @@ static void avifile_write_header(muxer_t
 	  idxhdr[6] = 0;
 	  idxhdr[7] = 0;
 
-	  stream_write_buffer(muxer->stream,idxhdr,sizeof(idxhdr));
+	  stream_write_buffer(muxer->stream,(void*)idxhdr,sizeof(idxhdr));
 	  for (j=0; j<n; j++) {
 	      struct avi_odmlsuperidx_entry *entry = &si->superidx[j];
 	      unsigned int data[4];
@@ -414,7 +414,7 @@ static void avifile_write_header(muxer_t
 	      data[1] = le2me_32(entry->ofs >> 32);
 	      data[2] = le2me_32(entry->len);
 	      data[3] = le2me_32(entry->duration);
-	      stream_write_buffer(muxer->stream,data,sizeof(data));
+	      stream_write_buffer(muxer->stream,(void*)data,sizeof(data));
 	  }
       }
   }
@@ -642,13 +642,13 @@ static void avifile_odml_write_index(mux
 	si->superidx[j].ofs = stream_tell(muxer->stream);
 	si->superidx[j].duration = duration;
 
-	stream_write_buffer(muxer->stream, idxhdr,sizeof(idxhdr));
+	stream_write_buffer(muxer->stream, (void*)idxhdr,sizeof(idxhdr));
 	for (k=0; k<entries_per_subidx && idxpos<si->idxpos; k++) {
 	    unsigned int entry[2];
 	    entry[0] = le2me_32(si->idx[idxpos].ofs - start);
 	    entry[1] = le2me_32(si->idx[idxpos].len | si->idx[idxpos].flags);
 	    idxpos++;
-	    stream_write_buffer(muxer->stream, entry, sizeof(entry));
+	    stream_write_buffer(muxer->stream, (void*)entry, sizeof(entry));
 	}
      }
   }
--- MPlayer-1.5/libmpcodecs/ve_lavc.c.orig	2021-05-14 21:56:16.000000000 +0200
+++ MPlayer-1.5/libmpcodecs/ve_lavc.c	2025-08-14 21:58:16.152399227 +0200
@@ -441,7 +441,7 @@ static int config(struct vf_instance *vf
 	    mp_msg(MSGT_MENCODER,MSGL_ERR,"error parsing vrc_q\n");
             return 0;
         }
-        lavc_venc_context->rc_override=
+        lavc_venc_context->rc_override=(void*)
             av_reallocp_array(lavc_venc_context->rc_override, i+1, sizeof(*lavc_venc_context->rc_override));
         lavc_venc_context->rc_override[i].start_frame= start;
         lavc_venc_context->rc_override[i].end_frame  = end;
@@ -710,7 +710,7 @@ static double psnr(double d){
     return -10.0*log(d)/log(10);
 }
 
-static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts){
+static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double){
     AVFrame *pic= vf->priv->pic;
 
     pic->data[0]=mpi->planes[0];
--- MPlayer-1.5/libmpcodecs/ve_libdv.c.orig	2010-09-12 18:08:57.000000000 +0200
+++ MPlayer-1.5/libmpcodecs/ve_libdv.c	2025-08-14 21:58:42.945890093 +0200
@@ -88,7 +88,7 @@ static int query_format(struct vf_instan
     return 0;
 }
 
-static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts){
+static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double){
 
     dv_encode_full_frame(vf->priv->enc, mpi->planes,
 	(mpi->flags&MP_IMGFLAG_YUV) ? e_dv_color_yuv : e_dv_color_rgb,
--- MPlayer-1.5/libmpcodecs/ve_nuv.c.orig	2010-11-14 10:12:34.000000000 +0100
+++ MPlayer-1.5/libmpcodecs/ve_nuv.c	2025-08-14 21:58:59.405986871 +0200
@@ -120,7 +120,7 @@ static int query_format(struct vf_instan
   return 0;
 }
 
-static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts){
+static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double){
   uint8_t *header  = vf->priv->buffer;
   uint8_t* data = vf->priv->buffer + FRAMEHEADERSIZE;
   uint8_t* zdata = vf->priv->zbuffer + FRAMEHEADERSIZE;
--- MPlayer-1.5/libmpcodecs/ve_x264.c.orig	2021-01-23 19:22:09.000000000 +0100
+++ MPlayer-1.5/libmpcodecs/ve_x264.c	2025-08-14 21:59:20.506110931 +0200
@@ -62,7 +62,7 @@ typedef struct h264_module_t {
 static x264_param_t param;
 static int parse_error = 0;
 
-static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts);
+static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double);
 static int encode_frame(struct vf_instance *vf, x264_picture_t *pic_in);
 
 void x264enc_set_param(const m_option_t* opt, char* arg)
@@ -234,7 +234,7 @@ static int query_format(struct vf_instan
     return 0;
 }
 
-static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts)
+static int put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double)
 {
     h264_module_t *mod=(h264_module_t*)vf->priv;
     int i;
--- MPlayer-1.5/libmpcodecs/ve_xvid4.c.orig	2021-01-23 19:22:09.000000000 +0100
+++ MPlayer-1.5/libmpcodecs/ve_xvid4.c	2025-08-15 06:23:10.998215058 +0200
@@ -523,7 +523,7 @@ query_format(struct vf_instance *vf, uns
  *==========================================================================*/
 
 static int
-put_image(struct vf_instance *vf, mp_image_t *mpi, double pts)
+put_image(struct vf_instance *vf, mp_image_t *mpi, double pts, double)
 {
 	int size;
 	xvid_enc_stats_t stats;
