--- MPlayer-1.0pre6/configure.orig	2005-02-23 15:24:58.000000000 +0100
+++ MPlayer-1.0pre6/configure	2005-02-23 21:16:47.155134160 +0100
@@ -4495,7 +4495,7 @@
      ( jackd --version | grep version | awk '{ print $3 }' ) >> "$TMPLOG" 2>&1 ; then
 
 cat > $TMPC << EOF
-#include <jack/jack.h>
+#include <bio2jack.h>
 int main(void) { JACK_Init(); return 0; }
 EOF
         # This test only checks the minor version number.
