--- MPlayer-1.0pre6/configure.orig	2005-02-23 15:24:58.000000000 +0100
+++ MPlayer-1.0pre6/configure	2005-02-23 21:16:47.155134160 +0100
@@ -4495,7 +4495,7 @@
      ( jackd --version | grep version | awk '{ print $3 }' ) >> "$TMPLOG" 2>&1 ; then
 
 cat > $TMPC << EOF
-#include <jack/jack.h>
+#include <bio2jack.h>
 int main(void) { JACK_Init(); return 0; }
 EOF
         # This test only checks the minor version number.
--- MPlayer-1.0pre7try2/libao2/ao_jack.c.orig	2005-03-25 16:11:13.000000000 +0100
+++ MPlayer-1.0pre7try2/libao2/ao_jack.c	2005-11-28 13:07:32.570007000 +0100
@@ -18,29 +18,12 @@
 #include "libaf/af_format.h"
 #include "mp_msg.h"
 
-//#include "bio2jack.h"
+#include <bio2jack.h>
 
 static int driver = 0;
 long latency = 0;
 long approx_bytes_in_jackd = 0;
 
-//bio2jack stuff:
-#define ERR_SUCCESS						0
-#define ERR_OPENING_JACK				1
-#define ERR_RATE_MISMATCH				2
-#define ERR_BYTES_PER_FRAME_INVALID		3
-enum status_enum { PLAYING, PAUSED, STOPPED, CLOSED, RESET };
-void JACK_Init(void);
-int  JACK_Open(int* deviceID, unsigned int bits_per_sample, unsigned long *rate, int channels);
-int  JACK_Close(int deviceID); /* return 0 for success */
-void JACK_Reset(int deviceID); /* free all buffered data and reset several values in the device */
-long JACK_Write(int deviceID, char *data, unsigned long bytes); /* returns the number of bytes written */
-long JACK_GetJackLatency(int deviceID); /* return the latency in milliseconds of jack */
-int  JACK_SetState(int deviceID, enum status_enum state); /* playing, paused, stopped */
-int  JACK_SetAllVolume(int deviceID, unsigned int volume);
-int  JACK_SetVolumeForChannel(int deviceID, unsigned int channel, unsigned int volume);
-void JACK_GetVolumeForChannel(int deviceID, unsigned int channel, unsigned int *volume);
-//
 
 
 static ao_info_t info =
